File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[] query) {
        List<Integer> list = new ArrayList<>();
        for (int num : arr) {
            list.add(num);
        }
        
        for (int i = 0; i < query.length; i++) {
            if (i % 2 == 0) {
                // 짝수 인덱스: query[i] 이후 제거
                while (list.size() > query[i] + 1) {
                    list.remove(list.size() - 1);
                }
            } else {
                // 홀수 인덱스: query[i] 이전 제거
                int removeCount = query[i];
                while (removeCount > 0) {
                    list.remove(0);
                    removeCount--;
                }
            }
        }
        
        return list.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.js
function solution(arr, query) {
    let list = [...arr];

    for (let i = 0; i < query.length; i++) {
        if (i % 2 === 0) {
            // even index: remove after query[i]
            list = list.slice(0, query[i] + 1);
        } else {
            // odd index: remove before query[i]
            list = list.slice(query[i]);
        }
    }

    return list;
}